pkg/scheduler/objects/node_iterator.go

```diff
@@ -19,103 +19,44 @@
 package objects
 
 import (
-	"math/rand"
+	"github.com/google/btree"
 )
 
 // NodeIterator iterates over a list of nodes based on the defined policy
 type NodeIterator interface {
-	// returns true if there are more values to iterate over
-	HasNext() bool
-	// returns the next node from the iterator
-	Next() *Node
-	// reset the iterator to a clean state
+	// Reset the iterator to a clean state
```

Comment:
Remove as this is now a no-op for the only implementation. Calling it has no effect on the result of ForEachNode().